File: AtvWendel04/AtvWendel04_PHP/Pessoa.php
<?php
class Pessoa {
    private $name;
    private $birth;
    public $surname = "";
    public $age;

    public function __construct($name, $birthDate) {
        $this->name = $name;
        $this->birth = new DateTime($birthDate);
        $this->calcAge();
    }

    private function calcAge() {
        $now = new DateTime();
        $interval = $this->birth->diff($now);
        $this->age = $interval->y;
    }

    public function getAge() {
        return $this->age;
    }

    public function hello() {
        $times = "";
        $hour = (int)date("H");

        if ($hour >= 5 && $hour <= 11) {
            $times = "dia";
        } else if ($hour >= 12 && $hour <= 18) {
            $times = "tarde";
        } else if ($hour >= 19 && $hour <= 23) {
            $times = "noite";
        } else if ($hour >= 0 && $hour <= 4) {
            $times = "madrugada";
        }

        return "Boa $times, me chamo $this->name, mas meu apelido é $this->surname, tenho {$this->getAge()} anos.";
    }
}
?>

File: AtvWendel04/AtvWendel04_Python/Pessoa.py
from datetime import datetime

class Pessoa:
    def __init__(self, name, birth_date):
        self.name = name
        self.birth = datetime.strptime(birth_date, '%Y-%m-%d')
        self.calc_age()

    def calc_age(self):
        now = datetime.now()
        age = now.year - self.birth.year - ((now.month, now.day) < (self.birth.month, self.birth.day))
        self.age = age

    def get_age(self):
        return self.age

    def hello(self):
        times = ""
        hour = datetime.now().hour

        if 5 <= hour <= 11:
            times = "dia"
        elif 12 <= hour <= 18:
            times = "tarde"
        elif 19 <= hour <= 23:
            times = "noite"
        elif 0 <= hour <= 4:
            times = "madrugada"

        return f"Boa {times}, me chamo {self.name}, mas meu apelido é {self.surname}, tenho {self.get_age()} anos."
